submissions/zalevskyi/dom/js/app.js

```diff
@@ -0,0 +1,110 @@
+        img_src: './img/Reyna.png',
+        img_alt: 'Reyna portrait'
+    },
+    'victor' : {
+        type: 'governor',
+        title: 'The Castellan',
+        description: 'Military focused, once established in a city, Victor is instrumental in strengthening, preserving, and defending it from foreign threats.',
+        context: 'A Castellan is a military officer responsible for directing the day-to-day affairs of a castle or fortress and the territory immediately around it. As castles were massive investments of treasure and important points of military power, the castellan had to be both competent and politically reliable. In times of war, a competent castellan could determine the fate of campaigns, and with that, the historical outcome.',
+        img_src: './img/Victor.png',
+        img_alt: 'Victor portrait'
+    }
+}
+const ARTICLE_INNER_HTML = {
+    'overview': `<h1 id='title' class='text_heading_1'></h1><p id='description' class='text'></p>`,
+    'governor': `<h1 id='title' class='text_heading_1'></h1>
+                 <img id='portrait' alt='' src='' class='portrait'>
+                 <h2 class='text_heading_2 description_heading'>Description</h2>
+                 <p id='description' class='text description'></p>
+                 <h2 class='text_heading_2 context_heading'>Historical Context</h2>
+                 <p id='context' class='text context'></p>`
+}
+Object.freeze(WIKI)
+Object.freeze(ARTICLE_INNER_HTML)
+function updateCommon(name) {
+    document.getElementById('title').textContent = `${name} - ${WIKI[name].title}`
+    document.getElementById('description').textContent = WIKI[name].description
+}
+function updateGovernor(name) {
+    document.getElementById('context').textContent = WIKI[name].context
+    document.getElementById('portrait').src = WIKI[name].img_src
+    document.getElementById('portrait').alt = WIKI[name].img_alt
+}
+document.getElementById('menu').addEventListener('click', ({target}) => {
+    const target_before = document.querySelector('.menu_item_selected')
+    if (target!==target_before) {
+        const article = document.getElementById('article')
+        const name = target.textContent
+        const name_before = target_before.textContent
+        target_before.classList.toggle('menu_item_selected')
+        target.classList.toggle('menu_item_selected')
+        article.classList.add('hidden')
+        if (WIKI[name]['type']!==WIKI[name_before]['type']) {
+            article.innerHTML=ARTICLE_INNER_HTML[WIKI[name]['type']]
+        }
+        updateCommon(name)
+        if (WIKI[name]['type']==='governor') updateGovernor(name)
+        article.classList.remove('hidden')
+    }
+    document.querySelector('.main').classList.toggle('hidden_small')
+    document.querySelector('.nav').classList.toggle('hidden_small')
+})
+document.getElementById('menu').addEventListener('click', e => e.preventDefault())
```

Comment:
no need to add so much event listeners. It can descrease the performance of your app.
Let's move this operation to the previous event listener